### PR TITLE
[Merged by Bors] - chore: make CanonicallyLinearOrdered[Add]CommMonoid extend more

### PR DESCRIPTION
**Pull request by madvorak**
Originally:
```
class CanonicallyLinearOrderedAddCommMonoid (α : Type*)
  extends CanonicallyOrderedAddCommMonoid α, LinearOrder α

class CanonicallyLinearOrderedCommMonoid (α : Type*) extends CanonicallyOrderedCommMonoid α,
  LinearOrder α
```
Now:
```
class CanonicallyLinearOrderedAddCommMonoid (α : Type*)
  extends CanonicallyOrderedAddCommMonoid α, LinearOrderedAddCommMonoid α

class CanonicallyLinearOrderedCommMonoid (α : Type*)
  extends CanonicallyOrderedCommMonoid α, LinearOrderedCommMonoid α
```

It gives the same union of properties as before, but now we have two more conversions for free:
`CanonicallyLinearOrderedAddCommMonoid ` -> `LinearOrderedAddCommMonoid `
`CanonicallyLinearOrderedCommMonoid ` -> `LinearOrderedCommMonoid `

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Order/Monoid/Canonical/Defs.lean [hunk 1]
```diff
@@ -393,4 +393,8 @@ theorem bot_eq_one' : (⊥ : α) = 1 :=
 #align bot_eq_one' bot_eq_one'
 #align bot_eq_zero' bot_eq_zero'
 
+@[to_additive]
+instance CanonicallyLinearOrderedCommMonoid.linearOrderedCommMonoid : LinearOrderedCommMonoid α :=
```
Comment (eric-wieser) [inline Mathlib/Algebra/Order/Monoid/Canonical/Defs.lean:397]:
```suggestion
instance CanonicallyLinearOrderedCommMonoid.toLinearOrderedCommMonoid : LinearOrderedCommMonoid α :=
```

Comment (eric-wieser) [inline Mathlib/Algebra/Order/Monoid/Canonical/Defs.lean:397]: Can you just change the`extends` clause instead?

Comment (madvorak) [reply Mathlib/Algebra/Order/Monoid/Canonical/Defs.lean]: I can change the definitions to
```
/-- A canonically linear-ordered additive monoid is a canonically ordered additive monoid
    whose ordering is a linear order. -/
class CanonicallyLinearOrderedAddCommMonoid (α : Type*)
  extends CanonicallyOrderedAddCommMonoid α, LinearOrderedAddCommMonoid α
#align canonically_linear_ordered_add_monoid CanonicallyLinearOrderedAddCommMonoid

/-- A canonically linear-ordered monoid is a canonically ordered monoid
    whose ordering is a linear order. -/
class CanonicallyLinearOrderedCommMonoid (α : Type*)
  extends CanonicallyOrderedCommMonoid α, LinearOrderedCommMonoid α
#align canonically_linear_ordered_monoid CanonicallyLinearOrderedCommMonoid
```
but then none of the `@[to_additive]` theorems works.
```
application type mismatch
  LinearOrderedAddCommMonoid.toAddCommMonoid
argument has type
  LinearOrderedCommMonoid α
but function has type
  [self : LinearOrderedAddCommMonoid α] → AddCommMonoid α
```

Comment (eric-wieser) [reply Mathlib/Algebra/Order/Monoid/Canonical/Defs.lean]: I think it's worth trying to diagnose why that happens

Comment (madvorak) [reply Mathlib/Algebra/Order/Monoid/Canonical/Defs.lean]: Do you know anyone who understands how `@[to_additive]` precisely works?

Comment (eric-wieser) [reply Mathlib/Algebra/Order/Monoid/Canonical/Defs.lean]: I think you just need to add `attribute [to_additive existing] LinearOrderedCommMonoid.toCommMonoid` or similar; there are other examples of this elsewhere.

File: Mathlib/Algebra/Order/Monoid/Canonical/Defs.lean [hunk 2]
```diff
@@ -334,20 +334,22 @@ end NeZero
...
-  LinearOrder α
+class CanonicallyLinearOrderedCommMonoid (α : Type*)
+  extends CanonicallyOrderedCommMonoid α, LinearOrderedCommMonoid α
 #align canonically_linear_ordered_monoid CanonicallyLinearOrderedCommMonoid
 
 section CanonicallyLinearOrderedCommMonoid
 
 variable [CanonicallyLinearOrderedCommMonoid α]
 
+attribute [to_additive existing] CanonicallyLinearOrderedCommMonoid.toLinearOrderedCommMonoid
```
Comment (eric-wieser) [inline Mathlib/Algebra/Order/Monoid/Canonical/Defs.lean:351]: Can you move this immediately before the relevant class?

Otherwise, thanks!

bors d+

**Comment by eric-wieser** [comment]
Please also fix the PR title to match the conventions, and write a short description

File: Mathlib/Algebra/Order/Monoid/Canonical/Defs.lean (hunk 2 above)
Comment (madvorak) [reply Mathlib/Algebra/Order/Monoid/Canonical/Defs.lean:351]: Right now it is immediately before the instance that requires it. Did you mean immediately AFTER the class?

Comment (eric-wieser) [reply Mathlib/Algebra/Order/Monoid/Canonical/Defs.lean:351]: Sorry, that is what I meant

**Pull request closed without merging**